src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4491,6 +4491,7 @@ private TypeSymbolWithAnnotations InferResultNullabilityOfBinaryLogicalOperator(
         public override BoundNode VisitAwaitExpression(BoundAwaitExpression node)
         {
             var result = base.VisitAwaitExpression(node);
+            CheckPossibleNullReceiver(node.Expression);
```

Comment:
What about the case where `GetAwaiter` is an extension method which takes a nullable receiver? What should the code do in that case and is it handled / tested here?

Reply:
I'll add a test, but I expect the behavior is the same as for invocations. Namely, we still warn. That was an explicit decision in a previous PR, but we could adjust later.